### PR TITLE
Address missing plans for people under 18.

### PR DESCRIPTION
**Pull request by TreyE**
ME-180642794

### CODE REVIEW

**Review by markgoho — CHANGES_REQUESTED** [review]
@TreyE can we add a spec here that would have failed prior to your fix being merged?

File: app/models/hbx_enrollment.rb [hunk 1]
```diff
@@ -1523,9 +1523,16 @@ def decorated_elected_plans(coverage_kind, market = nil)
...
-    elected_plans.reject!(&:allows_child_only_offering?) if ::EnrollRegistry.feature_enabled?(:exclude_child_only_offering) && any_member_greater_than_18?
-    elected_plans.reject!(&:allows_adult_and_child_only_offering?) if ::EnrollRegistry.feature_enabled?(:exclude_adult_and_child_only_offering) && !any_member_greater_than_18?
+  def filtered_elected_plans(elected_plans, coverage_kind)
+    # TODO: Address this more structurally later.  The logic for pediatric
+    #       dental plan eligibilty is not encoded in a way that interacts
+    #       intuitively with the two available values for whether a plan is
+    #       'child only' or 'adult and child only' on the child_only_offering
+    #       property of BenefitMarkets::Products::Product.
+    if coverage_kind == 'dental'
+      elected_plans.reject(&:allows_child_only_offering?) if ::EnrollRegistry.feature_enabled?(:exclude_child_only_offering) && any_member_greater_than_18?
```
Comment (nisanthyaganti9) [inline app/models/hbx_enrollment.rb:1533]: this should to be `reject!` instead of `reject`

Comment (TreyE) [reply app/models/hbx_enrollment.rb:1533]: Yup

**Review by nisanthyaganti9 — DISMISSED** [review]

**Review by TreyE — COMMENTED** [review]
I approve this peer review with the spec additions, but I need someone to mark it approved as I originated it and can't.

**Review by markgoho — APPROVED** [review]

**Pull request merged**